src/settingsview.cpp

```diff
@@ -90,6 +139,11 @@ void SettingsView::onDownloadDirChanged(const QString &dir)
     ui->downloadDirPath->setText(dir);
 }
 
+void SettingsView::onMonitorDirChanged(const QString &dir)
+{
+    ui->monitorDirPath->setText(dir);
+}
```

Comment:
There is incoherence between treatment of `ui->monitorClear` and `ui->monitorDirPath`. The former is updated in `SettingsView::clearMonitorDir()` and `SettingsView::browseMonitorDir()` but  omitted from this slot, while the opposite is true for `ui->monitorDirPath`.

You should localize all updates to the GUI side of monitor dir either in `SettingsView::clearMonitorDir()` and `SettingsView::browseMonitorDir()` (but then the `SettingsManager::monitorDirChanged()` signal and the `SettingsView::onMonitorDirChanged()` slot should be dropped), or in `SettingsView::onMonitorDirChanged()`.